src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -68,8 +67,6 @@ private function inlineArguments(ContainerBuilder $container, array $arguments,
             }
             if (is_array($argument)) {
                 $arguments[$k] = $this->inlineArguments($container, $argument);
-            } elseif ($argument instanceof ArgumentInterface) {
-                $argument->setValues($this->inlineArguments($container, $argument->getValues()));
```

Comment:
by removing this, we miss the opportunity to inline inside lazy arguments.
only the call to setValue should be removed in fact